File: lab-04/app/src/vecotr/Vector2D.java
package vecotr;

import static java.lang.Math.sqrt;

public class Vector2D implements IVector {

    private double x;
    private double y;
    public double getX() {        return x;    }
    public double getY() {        return y;    }

    public Vector2D(double x, double y) {

        this.x = x;
        this.y = y;
    }

    @Override
    public double abs() {
        return sqrt(x * x + y * y);
    }


    @Override
    public double cdot(IVector vec){
        return (vec.getX() * x + vec.getY() * y);
    }

    @Override
    public double[] getComponents() {
        double[] tab = { x,y };
        return tab;
    }
}

File: lab-04/app c#/app/vectors/Vector2d.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Drawing;
using System.Threading.Tasks;

namespace app
{

    class Vector2D 
    {


        private double x;
        private double y;

        public double X { get => x; set => x = value; }
        public double Y { get => y; set => y = value; }

        public Vector2D(double x, double y)
        {
            this.X = x;
            this.Y = y;
        }       

                                                                                                                               
        public double abs()
        {
            return Math.Sqrt(X * X + Y * Y);
        }
        public double cdot(Vector2D vector)
        {
            return (vector.X * X + vector.Y * Y);
        }
        public double[] getComponents()
        {
            double[] tab = { X, Y };
            return tab;
        }
        public double getAngle()
        {
            Vector2D ox = new Vector2D(1,0);
            double cosA = this.cdot(ox) / (this.abs() * ox.abs());
            return (180 / Math.PI) *  cosA;
        }

        public void chgangeDirection(bool b = false)
        {
            double temp = x;
            x = y;
            y = temp;
        }

    }




}
